system/doc/reference_manual/processes.xml

```diff
@@ -109,23 +185,508 @@ spawn(Module, Name, Args) -> pid()
+	released. With <i>directly visible Erlang resources</i> we here mean
+	all resources made available by the language excluding resources held
+	by heap data, dirty native code execution and the process identifier of
+	the terminating process. Examples of <i>directly visible Erlang
+	resources</i> are <seeguide marker="#registered-processes">registered
+	name</seeguide> and <seeerl marker="stdlib:ets">ETS</seeerl> tables.
+      </p>
+      <section>
+	<title>The Excluded Resources</title>
+	<p>
+	  The process identifier of the process cannot be released for
+	  reuse until everything regarding the process has been released.
+	</p>
+	<p>
+	  A process executing dirty native code in a NIF when it receives
+	  an exit signal will be set into an exiting state even though it is
+	  still executing dirty native code. <i>Directly visible Erlang
+	  resources</i> will be released, but the runtime system cannot
+	  force the native code to stop executing. The runtime system tries
+	  to prevent the execution of the dirty native code from effecting
+	  other processes by, for example, disabling functionality such as
+	  <seecref marker="erts:erl_nif#enif_send"><c>enif_send()</c></seecref>
+	  when used from a terminated process, but if the NIF is not well
+	  behaved it can still effect other processes. A well behaved dirty
+	  NIF should test if
+	  <seecref marker="erts:erl_nif#enif_is_current_process_alive">the
+	  process it is executing in has exited</seecref>, and if so stop
+	  executing.
+	</p>
+	<p>
+	  In the general case, the heap of a process cannot be removed before
+	  all signals that it needs to send have been sent. Resources held
+	  by heap data are the memory blocks containing the heap, but also
+	  include things referred to from the heap such as off heap binaries,
+	  and resources held via NIF
+	  <seecref marker="erts:erl_nif#resource_objects">resource
+	  objects</seecref> on the heap.
+	</p>
+      </section>
+    </section>
+
+    <section>
+      <marker id="signal-delivery"/>
+      <title>Delivery of Signals</title>
+      <p>
+	The amount of time that passes between the time a signal is sent and
+	the arrival of the signal at the destination is unspecified but
+	positive. If the receiver has terminated, the signal does not arrive,
+	but it can trigger another signal. For example, a <c>link</c> signal
+	sent to a non-existing process triggers an <c>exit</c> signal, which
+	is sent back to where the <c>link</c> signal originated from. When
+	communicating over the distribution, signals can be lost if the
+	distribution channel goes down.
+      </p>
+
+      <p>
+	The only signal ordering guarantee given is the following: if an
+	entity sends multiple signals to the same destination entity, the
+	order is preserved; that is, if <c>A</c> sends a signal <c>S1</c> to
+	<c>B</c>, and later sends signal <c>S2</c> to <c>B</c>, <c>S1</c> is
+	guaranteed not to arrive after <c>S2</c>.
```

Comment:
Should we also mention that in the scenario above, it is not guaranteed that S1 will arrive before S2? that is that S1 can be lost?